features/storage/blockdevice/BlockDevice.h

```diff
@@ -226,8 +226,8 @@ class BlockDevice
     bool is_valid_erase(bd_addr_t addr, bd_size_t size) const
     {
         return (
-            addr % get_erase_size() == 0 &&
-            size % get_erase_size() == 0 &&
+            addr % get_erase_size(addr) == 0 &&
+            size % get_erase_size(addr) == 0 &&
```

Comment:
You check that 'size' is aligned with current 'addr' but a large 'size' might take you to a 'dest' address in a different sector with a different 'erase-size' than 'addr'.
Basically I think you need to verify that:
( (addr+size) % get_erase_size(addr+size-1) == 0 )

Reply:
You are right about my code, but your solution is also problematic, as it only checks one sector. The right solution would be to iterate on the remaining sectors. Will fix that.

Reply:
You are right about the problem in my code, but your solution probably won't work either (as both only cover one sector). Right solution is probably an iteration over the remaining of the sectors. I'll fix it.

Reply:
@davidsaada - My thinking is that if the 'start' delete address is aligned to its sector, and the 'end' delete address is aligned to its own sector - then every sector in between them will be fully erased - so there's no need to iterate. What do you think?

Reply:
Fixed. Please review.